includes/class-newspack-ads-model.php

```diff
@@ -327,6 +327,8 @@ public static function sync_gam_settings( $serialised_ad_units = null, $settings
 			$serialised_ad_units = Newspack_Ads_GAM::get_serialised_gam_ad_units();
 		}
 		if ( null === $settings ) {
+			// Update targeting keys when fetching settings from GAM.
+			Newspack_Ads_GAM::update_custom_targeting_keys();
```

Comment:
As with other interactions with the external service (GAM), this should all be in a `try/catch` block. Apart from that, it would be nice if the user was notified about this, as well as any errors, should they arise.

Reply:
Good call! I've added `try/catch` for the sync method, which returns `WP_Error` that is handled by the `get_ad_units` method.

It required changes to the main plugin so it properly displays error notices and I've also moved the update execution to the `get_service()` method on the wizard class. This allows us to have better control over the state of the service on the UI.

To continue the review, also switch your `newspack-plugin` branch to this one: https://github.com/Automattic/newspack-plugin/pull/1146

You should remove the existing targeting keys so you can see the update in action.

In the case of targeting keys creation, the GAM card should display the following:
![image](https://user-images.githubusercontent.com/820752/133685386-637ddedf-71c9-4a0e-b964-dc99eeb8fbb8.png)

The link inside the notice goes to the Custom Targeting Keys managing dashboard on GAM.